Neos.Neos/Classes/View/FusionView.php

```diff
@@ -99,161 +44,128 @@ public function render(): string|ResponseInterface
-     *                                  otherwise the original output.
-     */
-    protected function parsePotentialRawHttpResponse($output)
+    /** Set the Fusion path to use for rendering the node given in "value" */
+    public function setFusionPath(string $fusionPath): void
     {
-        if ($this->isRawHttpResponse($output)) {
-            return Message::parseResponse($output);
-        }
+        $this->fusionPath = $fusionPath;
+    }
 
-        return $output;
+    public function getFusionPath(): string
+    {
+        return $this->fusionPath;
     }
 
     /**
-     * Checks if the mixed input looks like a raw HTTTP response.
+     * Assigns the Node to this view. The key must be "value".
+     * The documentNode and siteNode as well the site are derived from this entry Node.
      *
-     * @param mixed $value
-     * @return bool
+     * @throws \Exception If 'value' is not a Node
      */
-    protected function isRawHttpResponse($value): bool
+    public function assign($key, $value): AbstractView
     {
-        if (is_string($value) && strpos($value, 'HTTP/') === 0) {
-            return true;
+        if ($key !== 'value') {
+            // noop -> we only allow "value"
+            return $this;
         }
 
-        return false;
-    }
+        // 'value' is expected to be a Node!
+        if (!$value instanceof Node) {
+            throw new Exception('FusionView needs a variable \'value\' set with a Node object.', 1329736456);
+        }
 
-    /**
-     * Is it possible to render $node with $his->fusionPath?
-     *
-     * @return boolean true if $node can be rendered at fusionPath
-     *
-     * @throws Exception
-     */
-    public function canRenderWithNodeAndPath()
-    {
-        $currentSiteNode = $this->getCurrentSiteNode();
-        $fusionRuntime = $this->getFusionRuntime($currentSiteNode);
+        // clear the cached runtime instance
+        unset($this->fusionRuntime);
 
-        return $fusionRuntime->canRender($this->fusionPath);
+        $this->fusionRenderingStuff = $this->renderingUtility->createFusionRenderingStuff(
+            $value,
+            $this->controllerContext->getRequest()
+        );
+        return $this;
     }
 
     /**
-     * Set the Fusion path to use for rendering the node given in "value"
+     * Renders the view
      *
-     * @param string $fusionPath
-     * @return void
+     * @return string|ResponseInterface The rendered view
+     * @throws \Exception If no node is given
+     * @api
      */
-    public function setFusionPath($fusionPath)
+    public function render(): string|ResponseInterface
     {
-        $this->fusionPath = $fusionPath;
-    }
+        $this->initializeFusionRuntime();
 
-    /**
-     * @return string
-     */
-    public function getFusionPath()
-    {
-        return $this->fusionPath;
-    }
+        $this->fusionRuntime->pushContextArray(
+            $this->fusionRenderingStuff->fusionContext->toContextArray()
```

Comment:
I would actually prefer to make this more flexible than before. Have `assign` work as intended and add the other assigned variables to the context array here.